cunumeric/deferred.py

```diff
@@ -3224,3 +3223,36 @@ def unpackbits(
         scale = tuple(8 if dim == axis else 1 for dim in range(src.ndim))
         task.add_constraint(p_out <= p_in * scale)  # type: ignore
         task.execute()
+
+    @auto_convert([1])
+    def _wrap(self, src: Any, new_len: int) -> None:
+        if src.base.kind == Future:
+            src = self._convert_future_to_store(src)
+        if self.base.kind == Future:
+            self = src._convert_future_to_store(self)
+
+        # first, we create indirect array with PointN type that
+        # (len,) shape and is used to copy data from original array
+        # to the target 1D wrapped array
+        N = src.ndim
+        pointN_dtype = self.runtime.get_point_type(N)
+        indirect = cast(
+            DeferredArray,
+            self.runtime.create_empty_thunk(
+                shape=(new_len,),
+                dtype=pointN_dtype,
+                inputs=[src],
+            ),
+        )
+
+        task = self.context.create_task(CuNumericOpCode.WRAP)
+        task.add_output(indirect.base)
+        task.add_scalar_arg(src.shape, (ty.int64,))
+        task.execute()
+
+        copy = self.context.create_copy()
+        copy.set_target_indirect_out_of_range(False)
+        copy.add_input(src.base)
+        copy.add_source_indirect(indirect.base)
+        copy.add_output(self.base)
+        copy.execute()
```

Comment:
If `self` started out as a Future-backed array, the `self` variable that you have now is no longer a pointer to the same data. You need to do something similar to what you did for `setitem:
```
            if lhs is not self:
                self.copy(lhs, deep=True)
```

Reply:
good point, thanks